### PR TITLE
update examples to use TS 4.1 hook names

### PR DESCRIPTION
**Pull request by phryneas**

### CODE REVIEW

**Comment by phryneas** [comment]
A bit of a shame that CSB fails on this now.
I've created an issue over at https://github.com/codesandbox/codesandbox-client/issues/5128

File: examples/react/src/app/services/counter.ts [hunk 1]
```diff
@@ -35,3 +35,5 @@ export const counterApi = createApi({
     }),
   }),
 });
+
+export const { useDecrementCountMutation, useGetCountQuery, useIncrementCountMutation } = counterApi.hooks;
```
Comment (phryneas) [inline examples/react/src/app/services/counter.ts:39]: @markerikson this is what you have been waiting for, haven't you? ;)

Comment (markerikson) [reply examples/react/src/app/services/counter.ts:39]: niiiiice :)

File: src/ts41Types.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    ? QueryHook<Definitions[K]>
+    : Definitions[K] extends MutationDefinition<any, any, any, any>
+    ? MutationHook<Definitions[K]>
+    : never;
+};
+
+type TS41HookName<
+  K extends string,
+  Definition extends EndpointDefinition<any, any, any, any>
+> = `use${Capitalize<K>}${Definition extends QueryDefinition<any, any, any, any> ? 'Query' : 'Mutation'}`;
```
Comment (markerikson) [inline src/ts41Types.ts:20]: 🤯

Comment (phryneas) [reply src/ts41Types.ts:20]: yap :D

**Comment by phryneas** [comment]
This is so heavily outdated by now that we'll need to redo that anyways. Closing ^^

**Pull request closed without merging**